### DISCUSSION TITLE
Protect Insert & Update on certain Columns with Row Level Security [RLS]

### DISCUSSION THREAD

**Discussion by intellecta-lk** [supabase/supabase#14508] [Questions]
table primary column is a `UUID`. I don't want any user writing over auto generated `UUID`.

so I created a policy to only allow insert if `ID` value sent by user is `NULL`.
```sql
CREATE POLICY "Prevent from over writing id column" ON "public"."test"
AS PERMISSIVE FOR INSERT
TO public
WITH CHECK (id = NULL)
```

But when User hasn't provided a `ID` values the RLS should be true and allow the insert to happen. But it Won't

**Comment by GaryAustin1** [comment]
With CHECK looks at the final data to be inserted so includes defaults.
The easiest way to deal with this in an insert trigger function and set the column in that.

**Reply by steve-chavez** [reply]
Also check this previous discussion https://github.com/orgs/supabase/discussions/656#discussioncomment-4974429
